src/dmqnode/storage/Ring.d

```diff
@@ -542,9 +542,7 @@ public class RingNode : StorageChannels
 
         override protected void recycleStorageEngine ( StorageEngine storage )
         {
-            auto ring = cast(Ring)storage;
-            assert(ring);
-            this.outer.storage_pool.recycle(ring);
+            this.outer.storage_pool.recycle(downcastAssert!(Ring)(storage));
```

Comment:
I presume `downcastAssert` is also consciously left as something that will not happen in production builds.

Reply:
Yes.